roadmap/implementors-guide/guide.md

```diff
@@ -916,21 +916,40 @@ Furthermore, the protocols by which subsystems communicate with each other shoul
+- a new parablock candidate is available
+- a peer has seconded a parablock candidate
+
+This module is only ever interested in parablocks assigned to the particular parachain which this validator is currently handling.
+
+New parablock candidates may arrive from a potentially unbounded set of collators. This subsystem chooses either 0 or 1 of them per relay parent to second. If it chooses to second a candidate, it sends an apropriate message to the **Candidate Backing** subsystem to generate an appropriate `Statement`.
+
+All parablocks which peers have seconded are also sent to the Candidate Backing subsystem for re-validation.
+As seconded peers are tallied, double-votes are detected. If found, a report is sent to the **Misbehavior Arbitration** subsystem.
+
+In the event that a parablock candidate proves invalid, this subsystem will receive a message back from the Candidate Backing subsystem indicating so. If that parablock candidate originated from a collator, this subsystem will blacklist that collator. If that parablock candidate originated from a peer, this subsystem generates a report for the **Misbehavior Arbitration** subsystem.
 
-It also detects double-vote misbehavior by validators as it imports votes, passing on the misbehavior to the correct reporter and handler.
 
-When run as a validator, this is the subsystem which actually validates incoming candidates.
+TODO: more details, protocol, etc
+
+### Candidate Backing Subsystem
+
+#### Description
+
+The Candidate Backing subsystem ensures at least one preliminary validator commits to each parablock's correctness. Parablocks for which no validator will assert correctness are discarded. If the block later proves invalid, the initial backers are slashable; this gives polkadot a rational threat model during subsequent stages.
+
+Its role is to produce backable candidates for inclusion in new relay-chain blocks. It does so by issuing signed [Statements](#Statement-type) and tracking received statements signed by other validators. Once enough statements are received, they can be combined into backing for specific candidates.
 
 #### Protocol
 
-This subsystem receives messages of the type [CandidateBackingSubsystemMessage](#Candidate-Backing-Subsystem-Message).
+The **Candidate Selection** subsystem is the primary source of non-overseer messages into this subsystem. That subsystem generates appropriate [`CandidateBackingSubsystemMessage`s](#Candidate-Backing-Subsystem-Message), and passes them to this subsystem.
+
+This subsystem validates the candidates and generates an appropriate `Statement`. All `Statement`s are then passed on to the **Statement Distribution** subsystem to be gossiped to peers. Instances of `Statement::Invalid` are also passed back to the Candidate Selection subsystem, so it can take action against the originator of the candidate. To reduce traffic, we do not send valid `Statement`s back to the Candidate Selection subsystem.
```

Comment:
`Statement::Invalid` contains a hash, which the Candidate Selection system would have to associate with a given collator. The job then needs to contain a map of all selected candidates and which collators put them forward.

Reply:
> The job then needs to contain a map of selected candidates and which collators put them forward.

Yes, I figured Candidate Selection was the logical place to keep such a map.

> Strictly speaking, we don't need to pass the statement, but only a message indicating that a candidate (by hash or full candidate) is invalid.

True, but I figured that reusing the `Statement` type was the fastest path to victory here. If Candidate Selection receives a different variant of that enum (due to a bug in dispatch code), it can and should just ignore it.

Reply:
> Strictly speaking, we don't need to pass the statement, but only a message indicating that a candidate (by hash or full candidate) is invalid.

I've changed that behavior; you're right, it makes more sense to add a variant to `CandidateSelectionSubsystemMessage` to handle that case than to pass along a full `Statement`.